### PR TITLE
BUG: score and Hessian for Tweedie models

### PR DESCRIPTION
**Pull request by kshedden**
The weighted residual approach to calculating the score and Hessian is incorrect for Tweedie EQS models, because the likelihood is not based on a proper density.  This arose in discussions relating to #7476, but also has other implications (e.g. using gradient-based fitting and IRLS fitting for Tweedie models yields different results).

The propopsed solution is to calculate the score and Hessian directly for models that are not based on proper likelihoods.

### CODE REVIEW

**Comment by kshedden** [comment]
The current state of this PR fixes the score, the hessian is WIP.

**Comment by josef-pkt** [comment]
Doesn't this break the current estimation for the one parameter linear exponential family case?

If this doesn't replicate the GLM results given a fully specified variance function, then this should not be in GLM.

**Comment by josef-pkt** [comment]
aside: AFAIK, the special function needed for the full tweedie loglikelihood has been merged to scipy.special a while ago.

**Comment by josef-pkt** [comment]
scikit learn optimizes the deviance in GLM, which includes tweedie
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_glm/glm.py#L294

**Comment by josef-pkt** [comment]
something is strange with the extended quasi likelihood.

https://github.com/statsmodels/statsmodels/blob/main/statsmodels/genmod/families/family.py#L1492

has an extra term `p * log(mu)` that has mu in it and so affects the mean parameter estimates which then differ, AFAICS, from argmax deviance (and argmax loglike ?).

Based on brief browsing this conference article https://core.ac.uk/download/pdf/6347266.pdf#page=11 first article equation (4)
has endog y instead of mu in it, which would make it independent of the mean estimates.

I haven't found anything else for extended qmle for tweedie.

**Comment by kshedden** [comment]
It appears that the root problem was a bug in the Tweedie quasi-likelihood function.  Special-casing of gradient and hessians for Tweedie EQL seems not to be needed.  It is not clear yet if this resolves stability issues with Tweedie elastic net.

**Comment by josef-pkt** [comment]
LGTM
It would be better to squash so we don't get the commit that changed the other parts, derivatives, ... in the blame and history

can be merged when green.

for elastic net, I guess we still need fixing the scale during the optimization in the options, as discussed in the original issue

**Comment by josef-pkt** [comment]
I guess this also means that optimization methods that use loglike like bfgs or lbfgs produce the same parameter estimates as those only using derivative information, irls and newton.

**Comment by diegodebrito** [comment]
@kshedden @josef-pkt 

Sorry for commenting on this after it was merged, but it might be relevant for issue: https://github.com/statsmodels/statsmodels/issues/8136

On reading Nelder and Pregibon, it doesn't seem like they used mu on the implementation. Notice how equation 9 has V(y) and not V(mu). This pull request was necessary since the implementation here was using mu incorrectly.

Given that, there is a section on the paper about adjustments of V(y) at the origin, otherwise the extended quasi-likelihood will be infinite. This seems to be related to the issue above.

Any thoughts?

**Comment by josef-pkt** [comment]
I looked at the section of Nelder and Pregibon for adjusting V(y) for y = 0
They don't give an explicit formula for adjustments for the tweedie case.

I guess the simplest solution would be to add c=3/8 (or something like that)  V(y+c)
or just replace the y in the `p * np.log(endog + c)`

(I just ran into the 3/8 again in tests for poisson rates where adding 3/8 is part of variance stabilizing transform 
https://github.com/statsmodels/statsmodels/blob/main/statsmodels/stats/rates.py#L105
)
